app/src/main/java/org/wikipedia/page/PageAvailableOfflineHandler.kt

```diff
@@ -16,19 +17,16 @@ object PageAvailableOfflineHandler {
         callback.onFinish(WikipediaApp.instance.isOnline || (page.offline && !page.saving))
     }
 
-    @SuppressLint("CheckResult")
-    fun check(pageTitle: PageTitle, callback: Callback) {
+    fun check(lifeCycleScope: CoroutineScope, pageTitle: PageTitle, callback: Callback) {
         if (WikipediaApp.instance.isOnline) {
             callback.onFinish(true)
             return
         }
-        CoroutineScope(Dispatchers.Main).launch(CoroutineExceptionHandler { _, exception ->
-            run {
-                callback.onFinish(false)
-                L.w(exception)
-            }
+        lifeCycleScope.launch(CoroutineExceptionHandler { _, exception ->
+            callback.onFinish(false)
+            L.w(exception)
         }) {
-            val readingListPage = withContext(Dispatchers.IO) { AppDatabase.instance.readingListPageDao().findPageInAnyList(pageTitle) }
+            val readingListPage = AppDatabase.instance.readingListPageDao().findPageInAnyList(pageTitle)
```

Comment:
I believe this might still need to be wrapped in `withContext(Dispatchers.IO)`. The function `findPageInAnyList()` doesn't suspend, so it will run on the default dispatcher of the scope (Main).

The way this is currently written, it doesn't need to be a coroutine (there's nothing that `suspend`s).